src/main/java/com/linkedin/replica/wall/commands/impl/AddPostCommand.java

```diff
@@ -22,34 +18,35 @@ public AddPostCommand(HashMap<String, Object> args, DatabaseHandler dbHandler){
 
 
     @Override
-    public Object execute() throws ParseException {
+    public Object execute() {
 
         // get database handler that implements functionality of this command
         WallHandler dbHandler = (WallHandler) this.dbHandler;
 
         // validate that all required arguments that are passed
-        validateArgs(new String[]{"authorId", "type", "companyId", "privacy", "text", "hashtags", "mentions", "likesCount", "images", "videos", "urls", "commentsCount", "shares", "isCompanyPost", "isPrior"});
+        validateArgs(new String[]{"authorId", "type", "text", "headLine", "isArticle"});
 
         // call dbHandler to get error or success message from dbHandler
-        DateFormat format = new SimpleDateFormat("EEE MMM dd yyyy hh:mm a", Locale.ENGLISH);
-        Post post;
-        Gson googleJson = new Gson();
-        String authorId = args.get("authorId").toString();
-        String type = args.get("type").toString();
-        String companyId = args.get("companyId").toString();
-        String privacy = args.get("privacy").toString();
-        String text = args.get("text").toString();
-        Date timestamp = format.parse(args.get("timestamp").toString());
-        ArrayList<String> hashtags = googleJson.fromJson((JsonArray) args.get("hashtags"), ArrayList.class);
-        ArrayList<String> mentions = googleJson.fromJson((JsonArray) args.get("mentions"), ArrayList.class);
-        int likesCount = (int) args.get("likesCount");
-        ArrayList<String> images = googleJson.fromJson((JsonArray) args.get("images"), ArrayList.class);
-        ArrayList<String> videos = googleJson.fromJson((JsonArray) args.get("videos"), ArrayList.class);
-        ArrayList<String> urls = googleJson.fromJson((JsonArray) args.get("urls"), ArrayList.class);
-        int commentsCount = (int) args.get("commentsCount");
-        boolean isCompanyPost = (boolean) args.get("isCompanyPost");
-        boolean isPrior = (boolean) args.get("isPrior");
-        post = new Post(authorId, type, companyId, privacy, text, hashtags, mentions, likesCount, images, videos, urls, commentsCount, timestamp, isCompanyPost, isPrior);
+        Gson gson = new Gson();
+        JsonObject request = (JsonObject) args.get("request");
+        String authorId = request.get("authorId").getAsString();
+        String type = request.get("type").getAsString();
+        String text = request.get("text").getAsString();
+        String headLine = request.get("headLine").getAsString();
+        Long timestamp = System.currentTimeMillis();
+        ArrayList<String> images = gson.fromJson(request.get("images").getAsJsonArray(), ArrayList.class);
```

Comment:
Check! can be null.
you can also check type : it resembles media type and customize an error for not finding the media required if null